src/Compatibility/Microsoft.DotNet.PackageValidation/ReadMe.md

```diff
@@ -0,0 +1,198 @@
+
+Another common issue in the .NET ecosystem is that binary breaking changes aren't well understood. Developers tend to think of "does the source still compile" as the bar to determine whether an API is compatible. However, at the API level certain changes that are fine in C# aren't compatible at the level of IL, for example, adding a defaulted parameter or changing the value of a constant.
+
+Package Validation tooling will allow library developers to validate that their packages are well-formed and they they didn't make unintentional breaking changes.
+
+# How To Add Package Validation To Your Projects
+
+Package Validation is currently being shipped as a msbuild sdk package which can be consumed in the csproj. It is set of tasks and targets that run after the dotnet pack
+The csproj will look like
+
+```C#
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>netstandard2.0;net6.0</TargetFrameworks>
+  </PropertyGroup>
+
+  <Sdk Name="Microsoft.DotNet.PackageValidation" Version="1.0.0-preview4.21121.3" />
+</Project>
+```
+
+## Scenarios and User Experience
+
+The first preview version of the package validation covers the basic scenarios for package validation. They are as follows:-
+
+## Validation Among Compatible Frameworks
+
+Package containing compatible frameworks need to ensure that code compiled against one can run against other. Examples of compatible pairs will be .NET Standard2.0 and .NET 6.0, .NET 5.0 and .NET 6.0 . In both of these cases, the users can compile against .NET Standard 2.0 or NET 5.0 and try to run on .NET 6.0, if the binaries are not compatible the users might end up with runtime errors.
+
+Package Validation will catch these errors at pack time. eg.
+
+Finley is working on a game which do a lot of string manipulation. They need to support both .NET Framework and .NET Core customers. They started with just targeting .NET Standard 2.0 but now they realize they want to take advantage of the spans in .NET Core 3.0 and above to avoid unnecessary string allocations. In order to do that, they are multi-targeting for NET Standard2.0 and .NET 6.0,.
+
+Finley has written the following code:
+```c#
+#if NET6_0_OR_GREATER
+    public void DoStringManipulation(ReadOnlySpan<char> input)
+    {
+        // use spans to do string operations.
+    }
+#else
+    public void DoStringManipulation(string input)
+    {
+        // Do some string operations.
+    }
+#endif
+```
+
+When Finley packs the project it fails with the following error:
+
+```c#
+error : .NETStandard,Version=v2.0 assembly api surface area should be compatible with net6.0 assembly surface area so we can compile against .NETStandard,Version=v2.0 and run on net6.0 .framework.
```

Comment:
NIT: might be useful to show the command that Finley wrote to the console (dotnet pack) so that users can follow along where are those errors coming from.

Reply:
The only issue there is that there will be other extra info like project paths and validation package path 
```
E:\A>dotnet pack
Microsoft (R) Build Engine version 16.10.0-preview-21181-07+073022eb4 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  Restored E:\A\A.csproj (in 222 ms).
  A -> E:\A\bin\Debug\netstandard2.0\A.dll
  A -> E:\A\bin\Debug\net6.0\A.dll
  Successfully created package 'E:\AllHands\CheckIfZero\bin\Debug\A.1.0.0.nupkg'.
C:\Users\.nuget\packages\microsoft.dotnet.packagevalidation\1.0.0-preview4\build\Microsoft.DotNet.PackageValidation.targets(28,5): error : .NETStandard,Version=v2.0 assembly api surface area should be compatible with net6.0 assembly surface area so we can compile against .NETStandard,Version=v2.0 and run on net6.0 .framework. [E:\A\A.csproj]
C:\Users\.nuget\packages\microsoft.dotnet.packagevalidation\1.0.0-preview4\build\Microsoft.DotNet.PackageValidation.targets(28,5): error : API Compatibility errors between lib/netstandard2.0/A.dll (left) and lib/net6.0/A.dll (right): [E:\AllHands\A\A.csproj]
C:\Users\.nuget\packages\microsoft.dotnet.packagevalidation\1.0.0-preview4\build\Microsoft.DotNet.PackageValidation.targets(28,5): error CP0002: Member 'A.B.DoStringManipulation(string)' exists on the left but not on the right [E:\A\A.csproj]
```
do u think it adds enough value ?